cmd/agentctl/main.go

```diff
@@ -517,6 +522,43 @@ func templateDryRunCmd() *cobra.Command {
 	return cmd
 }
 
+func testLogs() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "test-logs [config file]",
+		Short: "Collect logs but print entries instead of sending them to Loki.",
+		Long: `Starts Promtail using its '--dry-run' flag, which will only print logs instead of sending them to the remote server.
+		This can be useful for debugging and understanding how logs are being parsed.`,
+		Args: cobra.ExactArgs(1),
+
+		Run: func(_ *cobra.Command, args []string) {
+			file := args[0]
+
+			cfg := config.Config{}
+			err := config.LoadFile(file, false, &cfg)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "failed to validate config: %s\n", err)
+				os.Exit(1)
+			}
+
+			logger := log.NewLogfmtLogger(log.NewSyncWriter(os.Stdout))
```

Comment:
Do we want to reuse the logging parameters from the `config.Server` block here with the server.NewLogger helper?

If we prefer to have our own, I'd say maybe it's worth sending it over to stderr instead. Both for duplicating what the Agent is currently doing, as well as allowing a user to distinguish between the two streams and possibly redirect the dry-run logs to a file for further manipulation.

Reply:
Thanks for the review @tpaschalis. I thought about using the server logger, but find it a bit strange to re-use the config from the server, since this is a CLI tool and there is no concept of server. I think it separating these two streams makes sense. I can switch to stderr for now.

Does anybody else have opinion on this?

Reply:
uhm, although we are writing both to stdout and stderr in the command, so we are still mixing logs :thinking:

Reply:
Eh, more of a nitpicky comment, if someone is testing with a pre-existing config file, they'd probably also have the server block, the choice is yours. :)

Also, what I meant originally is that by splitting the streams we'd allow the user to redirect the stdout only to a file to verify that everything's working properly. Not sure if it's terribly useful though.

Reply:
discussed this, since promtail uses its own logger and it is [hardcoded](https://github.com/grafana/loki/blob/main/clients/pkg/promtail/client/logger.go#L56) to use stdout, the server logger config would not affect the output of promtail. We don't seem to log much more in the agent code itself for this code path, so probably using the server logger isn't needed.

We could use the ad-hoc logger to log all output for the command itself, but that would be inconsistent with the other commands.

I will change the only place where we write to stdout in the command to stderr, and set stderr for the agent logs as well (which as I understand is consistent with how we log when running the agent in normal mode).